test/custom_task_test.go

```diff
@@ -61,30 +62,45 @@ func TestCustomTask(t *testing.T) {
 			Spec: v1beta1.PipelineRunSpec{
 				PipelineSpec: &v1beta1.PipelineSpec{
 					Tasks: []v1beta1.PipelineTask{{
-						Name: "custom-task",
+						Name: "custom-task-ref",
 						TaskRef: &v1beta1.TaskRef{
 							APIVersion: apiVersion,
 							Kind:       kind,
 						},
+					}, {
+						Name: "custom-task-spec",
+						TaskSpec: &v1beta1.EmbeddedTask{
```

Comment:
Suggest adding a `Spec` nested here and validating that it's rendered to `Raw` as expected